packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormRecordCreate.tsx

```diff
@@ -0,0 +1,164 @@
+    form.setValue('objectName', props.action.settings.input.objectName);
+
+    for (const [property, value] of Object.entries(
+      props.action.settings.input.objectRecord,
+    )) {
+      form.setValue(property, value);
+    }
+  }, [props.action.settings.input, form]);
+
+  const selectedObjectMetadataItemNameSingular = form.watch('objectName');
+
+  const selectedObjectMetadataItem = activeObjectMetadataItems.find(
+    (item) => item.nameSingular === selectedObjectMetadataItemNameSingular,
+  );
+  if (!isDefined(selectedObjectMetadataItem)) {
+    throw new Error('Should have found the metadata item');
+  }
+
+  const editableFields = selectedObjectMetadataItem.fields.filter(
+    (field) =>
+      field.type !== FieldMetadataType.Relation &&
+      !field.isSystem &&
+      field.isActive,
+  );
+
+  const saveAction = useDebouncedCallback(
+    async (formData: SendEmailFormData) => {
+      if (props.readonly === true) {
+        return;
+      }
+
+      const { objectName: updatedObjectName, ...updatedOtherFields } = formData;
+
+      props.onActionUpdate({
+        ...props.action,
+        settings: {
+          ...props.action.settings,
+          input: {
+            type: 'CREATE',
+            objectName: updatedObjectName,
+            objectRecord: updatedOtherFields,
+          },
+        },
+      });
+    },
+    1_000,
+  );
+
+  useEffect(() => {
+    return () => {
+      saveAction.flush();
+    };
+  }, [saveAction]);
+
+  const handleSave = form.handleSubmit((formData: SendEmailFormData) =>
+    saveAction(formData),
+  );
+
+  return (
+    <WorkflowEditGenericFormBase
+      HeaderIcon={
+        <IconAddressBook
+          color={theme.font.color.tertiary}
+          stroke={theme.icon.stroke.sm}
+        />
+      }
+      headerTitle="Record Create"
+      headerType="Action"
+    >
+      <Controller
+        name="objectName"
+        control={form.control}
+        render={({ field, formState }) => (
+          <Select
+            dropdownId="workflow-edit-action-record-create-object-name"
+            label="Object"
+            fullWidth
+            disabled={formState.disabled}
+            value={field.value}
+            emptyOption={{ label: 'Select an option', value: '' }}
+            options={availableMetadata}
+            onChange={(updatedObjectName) => {
+              field.onChange(updatedObjectName);
+              handleSave();
+            }}
+          />
+        )}
+      />
+
+      {editableFields.map((field) => (
+        <Controller
+          key={field.id}
+          name={field.name}
+          control={form.control}
+          render={({ field: formField }) => (
+            <VariableTagInput
+              inputId={field.id}
+              label={field.label}
+              placeholder="Enter value (use {{variable}} for dynamic content)"
+              // @ts-expect-error Temporary fix
+              value={formField.value}
+              onChange={(value) => {
```

Comment:
that need to be more complex. It works well for non composite fields, but it does not for composite fields like LINKS, ADDRESS, EMAILS, PHONES, ...
For those, you need a recursive Input function rendering, please, check what have been done with `renderFields` function in `packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormServerlessFunction.tsx`
See figma for the design -> https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=45225-479230&node-type=frame&t=spN9UE00UdvOjP4X-0
![image](https://github.com/user-attachments/assets/908d83a2-6851-4773-9438-86443f52444e)

Reply:
Thanks for mentioning. @thomtrp told me to create a first basic version as the desired implementation for fields was under discussion. We'll see what has been decided and maybe improve the implementation in a second PR.

Reply:
@martmull @Devessier let's discuss the strategy for frontend on Monday. We will have to create new components for each fields. Having a common base will help creating those components in parallel. So yes, I would prefer having a simple version as a start 👌

Reply:
let's talk about that on today yes
Ok for basic version, but i suggest we don't introduce duplication
Let's do one basic version, used everywhere, and upgrade it in the future